pkg/controller/cassandrarestore/cassandrarestore_controller.go

```diff
@@ -0,0 +1,396 @@
+	if err != nil {
+		reqLogger.Info("Sidecar communication error checking running Operation", "OperationId", restoreId)
+		return errorfactory.New(errorfactory.SidecarNotReady{}, err, "sidecar communication error")
+	}
+	restoreStatus, err := sr.GetRestorebyId(restoreId)
+	if err != nil {
+		reqLogger.Info("Sidecar communication error checking running Operation", "OperationId", restoreId)
+		return errorfactory.New(errorfactory.SidecarNotReady{}, err, "sidecar communication error")
+	}
+
+	if err := UpdateRestoreStatus(r.client, restore, *restoreStatus, log); err != nil {
+		return errors.WrapIfWithDetails(err, "could not update status for restore", "restore", restore)
+	}
+
+	// Restore operation failed or canceled,
+	// TODO : reschedule it by marking restore Condition.State = RestoreRequired ?
+	if restore.Status.Condition.Type.IsInError() {
+		return errorfactory.New(errorfactory.SidecarOperationFailure{}, err, "Sidecar Operation failed", fmt.Sprintf("restore operation id : %s", restoreId))
+	}
+
+	// Restore operation completed successfully
+	if restore.Status.Condition.Type.IsCompleted()  {
+		return nil
+	}
+
+	// TODO : Implement timeout ?
+	
+	// restore operation still in progress
+	log.Info("Sidecar operation is still running", "restoreId", restoreId)
+	return errorfactory.New(errorfactory.SidecarOperationRunning{}, errors.New("sidecar restore operation still running"), fmt.Sprintf("restore operation id : %s", restoreId))
+}
+
+func (r *ReconcileCassandraRestore) updateAndFetchLatest(ctx context.Context, restore *api.CassandraRestore) (*api.CassandraRestore, error) {
+	typeMeta := restore.TypeMeta
+	err := r.client.Update(ctx, restore)
+	if err != nil {
+		return nil, err
+	}
+	restore.TypeMeta = typeMeta
+	return restore, nil
+}
+
+/*func (r *ReconcileCassandraRestore) checkFinalizers(ctx context.Context, reqLogger logr.Logger, cluster *api.CassandraCluster, instance *api.CassandraRestore) (reconcile.Result, error) {
+	// run finalizers
+	var err error
+	if util.StringSliceContains(instance.GetFinalizers(), restoreFinalizer) {
+		// remove finalizer
+		if err = r.removeFinalizer(ctx, instance); err != nil {
+			return common.RequeueWithError(reqLogger, "failed to remove finalizer from CassandraRestore", err)
+		}
+	}
+	return common.Reconciled()
+}
+
+func (r *ReconcileCassandraRestore) removeFinalizer(ctx context.Context, restore *api.CassandraRestore) error {
+	restore.SetFinalizers(util.StringSliceRemove(restore.GetFinalizers(), restoreFinalizer))
+	_, err := r.updateAndFetchLatest(ctx, restore)
+	return err
+}
+
+func (r *ReconcileCassandraRestore) addFinalizer(reqLogger logr.Logger, restore *api.CassandraRestore) {
+	reqLogger.Info("Adding Finalizer for the NifiUser")
+	restore.SetFinalizers(append(restore.GetFinalizers(), restoreFinalizer))
+	return
+}*/
+
+func (r *ReconcileCassandraRestore) listPods(namespace string, selector map[string]string) (*corev1.PodList, error) {
```

Comment:
probably something we could refactor between all controllers in utils

Reply:
Yes